src/main/cpp/events/sampleevent.cpp

```diff
@@ -296,7 +296,8 @@ void SampleEvent::setLoopStartOffset( int value )
 
 int SampleEvent::getEventLength()
 {
-    return ( _playbackRate == 1.f || _loopeable ) ? _eventLength : ( int )(( float ) _eventLength / _playbackRate );
+    return _eventLength;
+   // return ( _playbackRate == 1.f || _loopeable ) ? _eventLength : ( int )(( float ) _eventLength / _playbackRate );
 }
```

Comment:
Not sure what the initial logic here was, but there is no reason to scale this to ratio, the event length has nothing to do with the _sample length_ ! (needs updating of unit test prior to branch merge).